lib/simple_forms_api_submission/metadata_validator.rb

```diff
@@ -37,12 +37,16 @@ def self.validate_file_number(metadata)
     end
 
     def self.validate_zip_code(metadata, zip_code_is_us_based)
-      zip_code = metadata['zipCode'].dup.gsub(/[^0-9]/, '')
-      validate_presence_and_stringiness(zip_code, 'zip code')
+      zip_code = metadata['zipCode']
+      if zip_code_is_us_based
+        validate_presence_and_stringiness(zip_code, 'zip code')
+        zip_code = zip_code.dup.gsub(/[^0-9]/, '')
+        zip_code.insert(5, '-') if zip_code.match?(/\A[0-9]{9}\z/)
+        zip_code = '00000' unless zip_code.match?(/\A[0-9]{5}(-[0-9]{4})?\z/)
+      else
+        zip_code = '00000'
```

Comment:
for a country like canada, is it still intentional to send zeroes?

Reply:
> for a country like canada, is it still intentional to send zeroes?

To meet the metadata validation requirements for zip code values, yes.